### ISSUE TITLE
Clash between `OptionalUnit` and `ImplicitUnitReturnType`

### ISSUE THREAD

**Issue by jonapoul** [detekt/detekt#9422]
## Expected Behavior

Shouldn't need to disable/suppress rules to avoid logical clashes.

## Observed Behavior

I'm seeing a clash of functionality between the OptionalUnit and ImplicitUnitReturnType rules, where one asks us to explicitly declare a Unit return type and the other says that it shouldn't be declared. As a result I need to disable or suppress one or both rules.

## Steps to Reproduce

In build script:
```kotlin
detekt {
  allRules = true
}
```

In source:
```kotlin
// Raises ImplicitUnitReturnType
fun functionA() = functionC(param = "abc")

// Raises OptionalUnit
fun functionB(): Unit = functionC(param = "xyz")

private fun functionC(param: String) {
  println("Returns unit with $param")
}
```

## Your Environment

* Version of detekt used: 2.0.0-alpha5
* Version of Gradle used (if applicable): 9.5.1

**Comment by cortinico** [comment]
> I'm seeing a clash of functionality between the OptionalUnit and ImplicitUnitReturnType rules, where one asks us to explicitly declare a Unit return type and the other says that it shouldn't be declared. As a result I need to disable or suppress one or both rules.

Yeah those two rules clash with each other in some use cases. `ImplicitUnitReturnType` should potentially fire only on `public` and `protected` functions, not all of them

**Comment by 3flex** [comment]
I don't see these conflicting? The guidance in `ImplicitUnitReturnType` says "Use a block statement to make clear this function will never return a value", which doesn't trigger either rule.

**Comment by jonapoul** [comment]
If that's the case, IMO these rules should be merged and clarified to explicitly state that.
